### PR TITLE
Bug 3358: [TEST]Add Dockerfile and README for testing on container

### PR DESCRIPTION
**Pull request by ykubota**
We should add a container to run test easily.

http://icedtea.classpath.org/bugzilla/show_bug.cgi?id=3358

### CODE REVIEW

File: agent/test/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Test for HeapStats
+
+Test cases for HeapStats. These test cases could run on containers.
+
+## Usage
+
+### on container
+
+```
+docker build --pull -t heapstats/$(basename $PWD) -f ./Dockerfile(.el7) .
```
Comment (YaSuenag) [inline agent/test/README.md]: Please add description for Red Hat subscription.
I guess we should have valid Red Hat subscription(s) to use the docker registry in Red Hat.

Comment (YaSuenag) [inline agent/test/README.md]: Please change tag name.
If you want to use directory basename as the tag, please add the reason.

File: agent/test/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+```
+If run under proxy, run with `--build-arg http_proxy=<proxy> --build-arg https_proxy=<proxy> --build-arg _JAVA_OPTIONS="-Dhttps.proxyHost=<proxy host> -Dhttps.proxyPort=<proxy port>"`
+
+Once you have the image, you can easily run tests as below.
+
+```
+docker run --rm  -t heapstats/$(basename $PWD) ./<dirctory>/testcase.sh
```
Comment (YaSuenag) [inline agent/test/README.md]: Please change tag name. See above.

File: agent/test/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+# You need security option when you run race-condition/testcase.sh
+docker run --rm --security-opt seccomp=unconfined -t heapstats/$(basename $PWD) ./race-condition/testcase.sh
```
Comment (YaSuenag) [inline agent/test/README.md]: Please change tag name. See above.

File: agent/test/README.md (hunk 1 above)
Comment (ykubota) [reply agent/test/README.md]: Sure, I write a description.

Comment (ykubota) [reply agent/test/README.md]: ok. I will update tag names.

**Comment by ykubota** [comment]
@YaSuenag I updated README and comments of Dockerfle. Please review again

**Review by YaSuenag — APPROVED** [review]

**Pull request closed without merging**